src/Avalonia.Base/Threading/DispatcherPriority.cs

```diff
@@ -1,74 +1,121 @@
         /// </summary>
-        Input = 5,
+        public static readonly DispatcherPriority Background = new(1);
 
         /// <summary>
-        /// The job will be processed after layout and render but before input.
+        /// The job will be processed with the same priority as input.
         /// </summary>
-        Loaded = 6,
+        public static readonly DispatcherPriority Input = new(2);
 
         /// <summary>
-        /// The job will be processed with the same priority as render.
+        /// The job will be processed after layout and render but before input.
         /// </summary>
-        Render = 7,
+        public static readonly DispatcherPriority Loaded = new(3);
         
         /// <summary>
         /// The job will be processed with the same priority as render.
         /// </summary>
-        Layout = 8,
-        
+        public static readonly DispatcherPriority Render = new(5);
+
         /// <summary>
-        /// The job will be processed with the same priority as data binding.
+        /// The job will be processed with the same priority as render.
         /// </summary>
-        DataBind = 9,
+        public static readonly DispatcherPriority Layout = new(6);
 
         /// <summary>
-        /// The job will be processed with normal priority.
+        /// The job will be processed with the same priority as data binding.
         /// </summary>
-        Normal = 10,
+        [Obsolete("WPF compatibility")] public static readonly DispatcherPriority DataBind = MinValue;
 
         /// <summary>
         /// The job will be processed before other asynchronous operations.
         /// </summary>
-        Send = 11,
-        
+        public static readonly DispatcherPriority Send = new(7);
+
         /// <summary>
         /// Maximum possible priority
         /// </summary>
-        MaxValue = 11
+        public static readonly DispatcherPriority MaxValue = Send;
+
+        // Note: unlike ctor this one is validating
+        public static DispatcherPriority FromValue(int value)
+        {
+            if (value < MinValue.Value || value > MaxValue.Value)
+                throw new ArgumentOutOfRangeException(nameof(value));
+            return new DispatcherPriority(value);
+        }
+
+        public static implicit operator int(DispatcherPriority priority) => priority.Value;
+
+        public static implicit operator DispatcherPriority(int value) => FromValue(value);
+
+        /// <inheritdoc />
+        public bool Equals(DispatcherPriority other) => Value == other.Value;
+
+        /// <inheritdoc />
+        public override bool Equals(object? obj) => obj is DispatcherPriority other && Equals(other);
+
+        /// <inheritdoc />
+        public override int GetHashCode() => Value.GetHashCode();
+
+        public static bool operator ==(DispatcherPriority left, DispatcherPriority right) => left.Value == right.Value;
```

Comment:
Could you use a `record struct` to remove the need for some of this boilerplate?